src/query/plan/operator.cpp

```diff
@@ -4861,6 +4868,9 @@ class CallProcedureCursor : public Cursor {
         throw QueryRuntimeException("{}: {}", self_->procedure_name_, *result_->error_msg);
       }
       result_row_it_ = result_->rows.begin();
+      if (!result_->is_transactional) {
+        skip_rows_with_deleted_values();
+      }
```

Comment:
Just to double check here, we are not stuck here in infinite loop if everything in result->rows needs to be skipped, right? In next iter it will try to pull next record, and if that is empty it will return false, otherwise enter procedure execution again

Reply:
While I don’t think that this would result in an infinite loop, you made me check and I’ve noticed that, if `skip_rows_with_deleted_values` skips **all** rows, `Pull` should return `false`. I’ll add the handling of this case

Reply:
How it works if everything in `result->rows` needs to be skipped:
1. mark `stream_exhausted` as true
2. try to pull another result
3. If there are no results left to be pulled, the program exits the loop. If there are more results to be pulled, the program stays in the loop to run the procedure on them, then back to 1.

Since the input cursor will be exhausted at some point and return false upon `Pull`, no infinite loop should result.